conductr_cli/license_validation.py

```diff
@@ -0,0 +1,153 @@
+            license.post_license(args, license_file)
+
+        has_license_support, license_updated = license.get_license(args)
+        validate_license_data(conductr_version, nr_of_agent_instances, license_updated)
+
+
+def validate_license_data(conductr_version, nr_of_agent_instances, license):
+    validate_version(conductr_version, license)
+    validate_nr_of_agents(nr_of_agent_instances, license)
+    validate_expiry(license)
+    validate_grants(license)
+
+
+def validate_version(conductr_version, license_data):
+    versions = get_value(license_data, 'conductrVersions')
+    if versions:
+        if len(versions):
+            for version in versions:
+                if can_run_version(version, conductr_version):
+                    return
+
+            raise LicenseValidationError([
+                'Unable to run sandbox version {}'.format(conductr_version),
+                'The license allows for version {}'.format(', '.join(versions))
+            ])
+    return
+
+
+def validate_nr_of_agents(nr_of_agent_instances, license_data):
+    nr_of_allowed_agents = get_value(license_data, 'maxConductrAgents')
+    if not nr_of_allowed_agents:
+        raise LicenseValidationError([
+            'Unable to allocate {} agents'.format(nr_of_agent_instances),
+            'The license does not specify the number of allowed agents'
+        ])
+    elif nr_of_agent_instances > nr_of_allowed_agents:
+        raise LicenseValidationError([
+            'Unable to allocate {} agents'.format(nr_of_agent_instances),
+            'The license allows for {} agents'.format(nr_of_allowed_agents)
+        ])
+
+
+def validate_expiry(license_data):
+    expiry_date = get_value(license_data, 'expires')
+    if expiry_date:
+        days_to_expiry = license.calculate_days_to_expiry(expiry_date)
+        if days_to_expiry < 0:
+            raise LicenseValidationError([
+                'License has expired',
+                license.format_expiry(expiry_date)
+            ])
+
+
+def validate_grants(license_data):
+    grants = get_value(license_data, 'grants')
+    if not grants:
+        raise LicenseValidationError([
+            'License does not include grant for {}'.format(CONDUCTR_GRANT),
+            'The license does not have any grant declared'
+        ])
+    elif CONDUCTR_GRANT not in grants:
+        raise LicenseValidationError([
+            'License does not include grant for {}'.format(CONDUCTR_GRANT),
+            'The license has grants for {}'.format(', '.join(grants))
+        ])
+
+
+def get_value(license_data, key):
+    if license_data:
+        if key in license_data:
+            value = license_data[key]
+            if value:
+                return value
+
+    return None
+
+
+def can_run_version(version_pattern, version_number):
+    version_pattern_parts = version_pattern.split('.')
+    version_number_parts = version_number.split('.')
+
+    for idx, pattern in enumerate(version_pattern_parts):
```

Comment:
I don't think I can modify it as per your suggestion, since the `return int(pattern) == int(number_part)` statement shorts circuit the checks, and the remainder of the `version_pattern_parts` may not get compared with `version_number_parts`.

So for example if the input `version_pattern` is `2.1.*` and `version_number` is `2.2.0`, I'd expect this method to return `False`. With your suggestion the method will return `True` since the first part for`version_pattern` and `version_number` is both `2`.